src/scanner.cc

```diff
@@ -271,6 +275,14 @@ struct Scanner {
     }
   }
 
+  
+  // This function returns true if c is a valid starting character of a name/identifier
+  bool is_identifier_start_char(char c){
+    return (c == '_')
+        || ('a' <= c && c <= 'z')
+        || ('A' <= c && c <= 'Z');
```

Comment:
Characters in the ranges between 128 and 255 are valid in variables apparently. Let's check those too. The generated `src/parser.c` file does this: `(128 <= lookahead && lookahead <= 255))`. We'd need to change the parameter type to `int32_t` which is the type of `lexer->lookahead`.

Can we also add a test for this in one of the here doc tests? We can use a variable with a character like `$€`.